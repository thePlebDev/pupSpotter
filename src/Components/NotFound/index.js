import React from 'react';
import styled from 'styled-components'
import {Link} from 'react-router-dom'

const H1 = styled.h1`
  font-size:60px;
  color:#3f51b5;
  opacity:0.8;

`
const Text = styled.div`
  font-size:160px;
  color:white;
  font-weight:1000;
  letter-spacing:20px;

  background: url("https://images.unsplash.com/photo-1511497584788-876760111969?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2089&q=80") center no-repeat;
`



const NotFound = ()=>{

  return(
    <div style={{width:'100%'}}>
      <div style={{width:'60%',margin:'0 auto',textAlign:'center'}}>
        <H1> Page not found </H1>
        <Text>404</Text>
        <Link to='/'>
          <H1 style={{cursor:'pointer' }}>Home</H1>
        </Link>

      </div>
    </div>
  )
}

export default NotFound
