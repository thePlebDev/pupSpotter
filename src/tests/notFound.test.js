import React from 'react';
import NotFound from '../Components/NotFound';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import {BrowserRouter as Router} from 'react-router-dom';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';


Enzyme.configure({
  adapter: new EnzymeAdapter()
})


describe('testing the not found',()=>{
  it('creating the snapshot',()=>{
    const NotFoundComponent = renderer.create(<Router><NotFound/></Router>).toJSON()
    expect(NotFoundComponent).toMatchSnapshot();
  })
  it('should render correctly',()=>{
  const component = shallow(<NotFound/>)
  const wrapper = component.find('[data-testid="notFound"]')
  expect(wrapper.length).toBe(1)
  })
})
