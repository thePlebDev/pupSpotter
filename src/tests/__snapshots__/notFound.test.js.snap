// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the not found creating the snapshot 1`] = `
<div
  data-testid="notFound"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "margin": "0 auto",
        "textAlign": "center",
        "width": "60%",
      }
    }
  >
    <h1
      className="sc-AxjAm chZeGo"
    >
       Page not found 
    </h1>
    <div
      className="sc-AxirZ klSXQc"
    >
      404
    </div>
    <a
      href="/"
      onClick={[Function]}
    >
      <h1
        className="sc-AxjAm chZeGo"
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Home
      </h1>
    </a>
  </div>
</div>
`;
